File: app/src/main/java/com/renzam/libyoutube/Adapter.java
package com.renzam.libyoutube;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.lifecycle.Lifecycle;
import androidx.recyclerview.widget.RecyclerView;
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.YouTubePlayer;
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.YouTubePlayerView;
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.AbstractYouTubePlayerListener;



public class Adapter extends RecyclerView.Adapter<Adapter.ViewHolder> {

    Context context;
    private String[] videoIds;
    private Lifecycle lifecycle;



    public Adapter(Context context, String[] videoIds, Lifecycle lifecycle) {
        this.context = context;
        this.videoIds = videoIds;
        this.lifecycle = lifecycle;
    }

    @NonNull
    @Override
    public Adapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        YouTubePlayerView youTubePlayerView = (YouTubePlayerView) LayoutInflater.from(parent.getContext()).inflate(R.layout.viedios_list, parent, false);

        lifecycle.addObserver(youTubePlayerView);
        return new ViewHolder(youTubePlayerView);

    }

    @Override
    public void onBindViewHolder(@NonNull Adapter.ViewHolder holder, int position) {

        holder.cueVideo(videoIds[position]);

    }

    @Override
    public int getItemCount() {
        return videoIds.length;
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        private YouTubePlayerView youTubePlayerView;
        private YouTubePlayer youTubePlayer;
        private String currentVideoId;

        ViewHolder(YouTubePlayerView playerView) {
            super(playerView);
            youTubePlayerView = playerView;

            youTubePlayerView.addYouTubePlayerListener(new AbstractYouTubePlayerListener() {
                @Override
                public void onReady(@NonNull YouTubePlayer initializedYouTubePlayer) {
                    youTubePlayer = initializedYouTubePlayer;
                    youTubePlayer.cueVideo(currentVideoId, 0);

                    youTubePlayerView.getPlayerUiController();

                }
            });
        }
        void cueVideo(String videoId) {
            currentVideoId = videoId;

            if (youTubePlayer == null)
                return;

            youTubePlayer.cueVideo(videoId, 0);
 }
    }
}

File: app/src/main/java/com/renzam/libyoutube/Adapter.kt
package com.renzam.libyoutube

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.core.content.ContextCompat.startActivity
import androidx.lifecycle.Lifecycle
import androidx.recyclerview.widget.RecyclerView
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.YouTubePlayer
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.YouTubePlayerView
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.AbstractYouTubePlayerListener
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.YouTubePlayerFullScreenListener
import com.pierfrancescosoffritti.androidyoutubeplayer.core.ui.PlayerUiController
import androidx.core.content.ContextCompat.startActivity
import androidx.core.content.ContextCompat.startActivity

class Adapter(private val context: Context, private val videoIds: Array<String>, private val lifecycle: Lifecycle) : RecyclerView.Adapter<Adapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Adapter.ViewHolder {

        val youTubePlayerView = LayoutInflater.from(parent.context).inflate(R.layout.viedios_list, parent, false) as YouTubePlayerView

        lifecycle.addObserver(youTubePlayerView)
        return ViewHolder(youTubePlayerView)


    }

    override fun onBindViewHolder(holder: Adapter.ViewHolder, position: Int) {


            holder.cueVideo(videoIds[position])

    }

    override fun getItemCount(): Int {
        return videoIds.size
    }

    inner class ViewHolder(private val youTubePlayerView: YouTubePlayerView) : RecyclerView.ViewHolder(youTubePlayerView) {
        private var youTubePlayer: YouTubePlayer? = null
        private var currentVideoId: String? = null

        init {

            youTubePlayerView.addYouTubePlayerListener(object : AbstractYouTubePlayerListener() {
                override fun onReady(initializedYouTubePlayer: YouTubePlayer) {
                    youTubePlayer = initializedYouTubePlayer
                    youTubePlayer!!.cueVideo(currentVideoId!!, 0f)

                }
            })
            youTubePlayerView.addFullScreenListener(object : YouTubePlayerFullScreenListener {
                override fun onYouTubePlayerEnterFullScreen() {

                    youTubePlayerView.toggleFullScreen()

                    //Intent to direct Youtube FullScreen
                    val intent = Intent(Intent.ACTION_VIEW,
                    Uri.parse("http://www.youtube.com/watch?v=" + videoIds[position]))
                    intent.putExtra("force_fullscreen", true)
                    context.startActivity(intent)
                }
                override fun onYouTubePlayerExitFullScreen() {

                    Toast.makeText(context,"Full screen exited",Toast.LENGTH_SHORT).show()


                }
            })

        }

        fun cueVideo(videoId: String) {
            currentVideoId = videoId

            if (youTubePlayer == null)
                return

            youTubePlayer!!.cueVideo(videoId, 0f)

        }
    }
}
